toolchain/check/import_ref.cpp

```diff
@@ -1266,14 +1269,31 @@ class ImportRefResolver {
 
     SemIR::ClassId class_id = SemIR::ClassId::Invalid;
     if (!class_const_id.is_valid()) {
+      auto import_specific_id = SemIR::SpecificId::Invalid;
+      if (auto import_generic_class_type =
+              import_ir_.types().TryGetAs<SemIR::GenericClassType>(
+                  inst.type_id)) {
+        import_specific_id = import_generic_class_type->enclosing_specific_id;
+      }
+      auto specific_data = GetLocalSpecificData(import_specific_id);
+      if (HasNewWork()) {
+        // This is the end of the first phase. Don't make a new class yet if
+        // we already have new work.
```

Comment:
This is marked as the end of the first phase, but so is line 1289. Is this actually becoming four phases? Should the check on line 1288 just be removed, comparing with the interface version?

I'll be approving assuming 1288 should just be removed, please let me know if I'm misunderstanding.

(note, this inconsistency makes me wonder if there's a good way to factor towards more code sharing)

Reply:
Thanks, the second check is redundant and it looks like I forgot to delete it. Deleted now matching the other cases.

Yeah, I think it'd be good to look into refactoring function / class / interface handling.